modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,217 @@
+	exists, err := LoadBalancerExistsE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// LoadBalancerExistsE indicates whether the specified Load Balancer exists.
+func LoadBalancerExistsE(loadBalancerName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetLoadBalancerFrontendIPConfigNames gets a list of the Frontend IP Configuration Names for the Load Balancer.
+// This function would fail the test if there is an error.
+func GetLoadBalancerFrontendIPConfigNames(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) []string {
+	configName, err := GetLoadBalancerFrontendIPConfigNamesE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return configName
+}
+
+// GetLoadBalancerFrontendIPConfigNamesE ConfigNamesE gets a list of the Frontend IP Configuration Names for the Load Balancer.
+func GetLoadBalancerFrontendIPConfigNamesE(loadBalancerName string, resourceGroupName string, subscriptionID string) ([]string, error) {
+	lb, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the Frontend IP Configurations
+	lbProps := lb.LoadBalancerPropertiesFormat
+	feConfigs := *lbProps.FrontendIPConfigurations
+	if len(feConfigs) == 0 {
+		// No Frontend IP Configuration present
+		return nil, nil
+	}
+
+	// Get the names of the Frontend IP Configurations present
+	configNames := make([]string, len(feConfigs))
+	for i, config := range feConfigs {
+		configNames[i] = *config.Name
+	}
+
+	return configNames, nil
+}
+
+// GetIPOfLoadBalancerFrontendIPConfig gets the IP and IPType for the specified Load Balancer Frontend IP Configuration.
+// This function would fail the test if there is an error.
+func GetIPOfLoadBalancerFrontendIPConfig(t testing.TestingT, feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType) {
+	ipAddress, ipType, err := GetIPOfLoadBalancerFrontendIPConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ipAddress, ipType
+}
+
+// GetIPOfLoadBalancerFrontendIPConfigE gets the IP and IPType for the specified Load Balancer Frontend IP Configuration.
+func GetIPOfLoadBalancerFrontendIPConfigE(feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType, err1 error) {
+	// Get the specified Load Balancer Frontend Config
+	feConfig, err := GetLoadBalancerFrontendIPConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", -1, err
```

Comment:
NIT: It is almost always better to return a valid enum value if using enum types. Either define a new enum val (e.g., `IPTypeError`), or pick one to return as default.

Reply:
Udated